### PR TITLE
testing: e2e tests for soft opt out 

### PR DESCRIPTION
**Pull request by smarshall-spitzbart**
# Description

Adds e2e tests validating #765 

## Linked issues

Closes https://github.com/cosmos/interchain-security/issues/849

## Type of change

If you've checked more than one of the first three boxes, consider splitting this PR into multiple PRs!

- [ ] `Feature`: Changes and/or adds code behavior, irrelevant to bug fixes
- [ ] `Fix`: Changes and/or adds code behavior, specifically to fix a bug
- [ ] `Refactor`: Changes existing code style, naming, structure, etc.
- [x] `Testing`: Adds testing
- [ ] `Docs`: Adds documentation

## Regression tests

Existing happy path steps are still there

## New behavior tests
Two new steps functions were added, stepsRedelegateForOptOut and stepsDowntimeWithOptOut. 

## Versioning Implications

- [ ] This PR will affect [semantic versioning as defined for ICS](../CONTRIBUTING.md#semantic-versioning)

If the above box is checked, which version should be bumped?

- [ ] `MAJOR`: Consensus breaking changes to both the provider and consumers(s), including updates/breaking changes to IBC communication between provider and consumer(s)
- [ ] `MINOR`: Consensus breaking changes which affect either only the provider or only the consumer(s)
- [ ] `PATCH`: Non consensus breaking changes

## Targeting

Please select one of the following:

- [x] This PR is only relevant to main
- [ ] This PR is relevant to main, and should also be back-ported to ____ (ex: v1.0.0 and v1.1.0)
- [ ] This PR is only relevant to ____ (ex: v1.0.0, v1.1.0, and v1.2.0)

### CODE REVIEW

**Comment by smarshall-spitzbart** [comment]
@mpoke lets chat tomorrow about whether this PR validates that the default value of `"0.05"` is indeed used as the default value for the consumer

**Comment by mpoke** [comment]
> @mpoke lets chat tomorrow about whether this PR validates that the default value of `"0.05"` is indeed used as the default value for `SoftOptOutThreshold` on the consumer

I don't think it does. IIUC, both the consumer and provider in the tests are running the same version of ICS, the one where the provider sets the `SoftOptOutThreshold` to `0.05`. Could you try to run the same test, but with ICS v1.1 on the provider?

**Comment by smarshall-spitzbart** [comment]
@mpoke `GAIA_TAG=v9.0.2 make test-gaia-e2e-short-tagged` fails but this is expected. For features that we get out quickly like this, consumers will need to manually mutate the genesis JSON queried from provider to have the proper parameter set. Stride is doing this manual mutation as well

**Comment by smarshall-spitzbart** [comment]
@mpoke with https://github.com/cosmos/interchain-security/pull/857/commits/bfa4f1c465547c0eed473e62b1b0ab7e86a21dee, `GAIA_TAG=v9.0.2 make test-gaia-e2e-short-tagged` now passes 🚀. Instructions will need to be relayed to Neutron that they'll need to add `soft_opt_out_threshold = \"0.05\"` to the appropriate section of their genesis JSON

**Review by MSalopek — APPROVED** [review]
Approved!

I like that it's in happy path so we have an A/B test confirming that both downtime jailing scenarios work as expected.

**Review by jtremback — APPROVED** [review]
Approved, with one minor comment

File: tests/e2e/steps_start_chains.go [hunk 1]
```diff
@@ -147,6 +147,12 @@ func stepsStartConsumerChain(consumerName string, proposalIndex, chainIndex uint
 					{id: validatorID("alice"), stake: 500000000, allocation: 10000000000},
 					{id: validatorID("carol"), stake: 500000000, allocation: 10000000000},
 				},
+				// For consumers that're launching with the provider being on an earlier version
+				// of ICS before the soft opt-out threshold was introduced, we need to set the
+				// soft opt-out threshold to 0.05 in the consumer genesis to ensure that the
+				// consumer binary doesn't panic. Sdk requires that all params are set to valid
+				// values from the genesis file.
+				genesisChanges: ".app_state.ccvconsumer.params.soft_opt_out_threshold = \"0.05\"",
```
[the remark this answers is not in the document]
Comment (smarshall-spitzbart) [reply tests/e2e/steps_start_chains.go:155]: 0 is a valid value for that param, but setting it to 0 would make the test fail (just confirmed) and wouldn't validate the feature. It'd essentially mean 0% of validator set is allowed to opt out

**Pull request merged**